src/main/java/com/weather/assignment/initializer/WeatherDataInitializer.java

```diff
@@ -0,0 +1,18 @@
+package com.weather.assignment.initializer;
+
+import com.weather.assignment.service.WeatherService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.CommandLineRunner;
+
+public class WeatherDataInitializer implements CommandLineRunner {
+
+    @Autowired
+    WeatherService weatherService;
```

Comment:
- The fields should always be private unless it is absolutely necessary.
- Always use constructor injection.